src/AutoRest.CSharp/Common/Generation/Writers/RequestWriterHelpers.cs

```diff
@@ -334,21 +334,38 @@ private static void WriteSerializationFormat(CodeWriter writer, SerializationFor
         private static void WriteQueryParameter(CodeWriter writer, CodeWriterDeclaration uri, QueryParameter queryParameter)
         {
             string? delimiter = GetSerializationStyleDelimiter(queryParameter.SerializationStyle);
-            string method = delimiter != null
+            bool explode = queryParameter.Explode;
+            string method = delimiter != null && !explode
                 ? nameof(RequestUriBuilderExtensions.AppendQueryDelimited)
                 : nameof(RequestUriBuilderExtensions.AppendQuery);
 
             ReferenceOrConstant value = queryParameter.Value;
             using (WriteValueNullCheck(writer, value))
             {
-                writer.Append($"{uri}.{method}({queryParameter.Name:L}, ");
-                WriteConstantOrParameter(writer, value, enumAsString: true);
-                if (delimiter != null)
+                if (explode)
                 {
-                    writer.Append($", {delimiter:L}");
+                    writer.Append($"foreach(var param in ");
```

Comment:
```cs
var paramVariable = new CodeWriterDeclaration("param");
using (writer.Scope($"foreach (var {paramVariable:D} in ")) {
```

Reply:
@AlexanderSher: how do I `WriteConstantOrParameter(writer, value, enumAsString: true)` within this using foreach declaration?

Reply:
since your cleaner declaration is missing the iterable within the foreach statement and since WriteConstantOrParameter calls writer apis, I either wrap the same logic that results in a string or probably just declare the paramVariable but keep the foreach logic as it was

Reply:
Thanks for the discovery, filed: https://github.com/Azure/autorest.csharp/issues/1533